src/cli/mod.rs

```diff
@@ -0,0 +1,125 @@
+pub mod mirror;
+pub mod scan;
+
+use crate::{self as stellar_archivist, mirror_operation, scan_operation};
+use clap::{Parser, Subcommand};
+use std::ffi::OsString;
+use std::io;
+use thiserror::Error;
+use tracing::Level;
+use tracing_subscriber::FmtSubscriber;
+
+/// CLI errors
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Clap(#[from] clap::Error),
+
+    #[error(transparent)]
+    Io(#[from] io::Error),
+
+    #[error(transparent)]
+    ScanOperation(#[from] scan_operation::Error),
+
+    #[error(transparent)]
+    MirrorOperation(#[from] mirror_operation::Error),
+
+    #[error("{0}")]
+    Other(String),
+}
+
+impl From<stellar_archivist::Error> for Error {
+    fn from(err: stellar_archivist::Error) -> Self {
+        match err {
+            stellar_archivist::Error::Io(e) => Error::Io(e),
+            stellar_archivist::Error::ScanOperation(e) => Error::ScanOperation(e),
+            stellar_archivist::Error::MirrorOperation(e) => Error::MirrorOperation(e),
+            stellar_archivist::Error::Other(s) => Error::Other(s),
+        }
+    }
+}
+
+#[derive(Parser)]
+#[command(name = "stellar-archivist")]
+#[command(about = "Stellar History Archive tools and utilities", long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Commands,
+
+    /// Number of files to process concurrently
+    #[arg(short, long, global = true, default_value_t = 32)]
+    concurrency: usize,
+
+    /// Skip optional SCP files
+    #[arg(long, global = true)]
+    skip_optional: bool,
+
+    /// Enable debug logging
+    #[arg(long, global = true)]
+    debug: bool,
+
+    /// Enable trace logging
+    #[arg(long, global = true)]
+    trace: bool,
+
+    /// Maximum number of HTTP retry attempts
+    #[arg(long, global = true, default_value_t = 3)]
+    max_retries: u32,
+
+    /// Initial backoff in milliseconds for HTTP retries
+    #[arg(long, global = true, default_value_t = 100)]
+    initial_backoff_ms: u64,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Mirror files from source archive to destination
+    Mirror(mirror::MirrorCmd),
+    /// Scan archive and verify integrity
+    Scan(scan::ScanCmd),
+}
+
+/// Global arguments shared by all commands
+#[derive(Debug, Clone)]
+pub struct GlobalArgs {
+    pub concurrency: usize,
+    pub skip_optional: bool,
+    pub max_retries: u32,
+    pub initial_backoff_ms: u64,
+}
```

Comment:
Fyi instead of creating a GlobalArgs that you have to copy the options into, you can make this struct the definition of the options, and then specify the GlobalArgs as a single field on the Cmd, and add the `#[arg(flatten)]` so that the options flatten into the command.

For example: https://github.com/stellar/stellar-cli/blob/9d9f4b3464e2a75d34898e9a55fb28e3757d06cb/cmd/soroban-cli/src/commands/mod.rs#L78-L79